### PR TITLE
feat: Add aura script to fetch bveCVX % amounts per data source

### PR DESCRIPTION
**Pull request by SHAKOTN**
1. Add script to calculate % per each data source
2. Add simplejson lib to store info to file and make it Decimal compatible

### CODE REVIEW

File: scripts/aura.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from rewards.boost.boost_utils import get_bvecvx_lp_ratio
+from rewards.classes.Snapshot import Snapshot
+from rewards.snapshot.chain_snapshot import sett_snapshot
+from rewards.snapshot.claims_snapshot import claims_snapshot
+from rewards.snapshot.token_snapshot import fuse_snapshot_of_token
+
+console = Console()
+
+
+STARTING_BLOCK = 14848251
```
Comment (mitche50) [inline scripts/aura.py:25]: Maybe rename `AURA_SNAPSHOT_BLOCK` for clarity

File: scripts/aura.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+MAX_TOLERANCE_THRESHOLD = Decimal(0.00001)
+
+
+def snapshot_to_percentages(snapshot: Snapshot):
+    percentages = {}
+    console.print(snapshot.total_balance())
+    for address, number in snapshot:
+        percentage = snapshot.percentage_of_total(address) * 100
+        # Filter out very small amounts
+        if percentage > Decimal(0.0000001):
```
Comment (mitche50) [inline scripts/aura.py:35]: Is this limiter too small?  What does 0.00001% equate to in $ value -> those users will never be able to claim anything is my assumption.

File: scripts/aura.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+console = Console()
+
+
+STARTING_BLOCK = 14848251
+MAX_TOLERANCE_THRESHOLD = Decimal(0.00001)
+
+
+def snapshot_to_percentages(snapshot: Snapshot):
+    percentages = {}
+    console.print(snapshot.total_balance())
```
Comment (mitche50) [inline scripts/aura.py:31]: Extra print - not informative

File: scripts/aura.py (hunk 2 above)
Comment (SHAKOTN) [reply scripts/aura.py:35]: Could be, but leaving that for Trit to decide

File: scripts/aura.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        Then: $ python -m scripts.aura
+    """
+    chain = Network.Ethereum
+    all_claims = claims_snapshot(chain, STARTING_BLOCK)
+    fuse_bvecvx = fuse_snapshot_of_token(chain, STARTING_BLOCK, addresses.BVECVX)
+    lp_bvecvx = sett_snapshot(chain, STARTING_BLOCK, addresses.BVECVX_CVX_LP_SETT)
+    claimable_bvecvx = all_claims[addresses.BVECVX]
+    ratio = get_bvecvx_lp_ratio()
+    ppfs = get_bvecvx_lp_ppfs()
+    for addr, value in lp_bvecvx:
```
Comment (mitche50) [inline scripts/aura.py:68]: A comment for added clarity here would be helpful for people not familiar with what `boost_balance` does.

```
# For each address holding the LP, multiply their holdings by the current ratio of bveCVX : CVX and the pricePerFullShare of the vault to determine underlying holdings
```

File: scripts/aura.py (hunk 1 above)
Comment (SHAKOTN) [reply scripts/aura.py]: Done

File: scripts/aura.py (hunk 4 above)
Comment (SHAKOTN) [reply scripts/aura.py:69]: Done

**Review by mitche50 — APPROVED** [review]

File: scripts/aura.py (hunk 2 above)
Comment (jorijnsmit) [reply scripts/aura.py:34]: i would argue limit of .0000001 is actually too low, i think that we are including too many people. any way to check $ value dropped to tree for example (prob not) or any other meaningful threshold?

File: scripts/aura.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+
+AURA_SNAPSHOT_BLOCK = 14829454
+MAX_TOLERANCE_THRESHOLD = Decimal(0.00001)
+
+
+def snapshot_to_percentages(snapshot: Snapshot):
+    percentages = {}
+    for address, number in snapshot:
+        percentage = snapshot.percentage_of_total(address) * 100
```
Comment (jorijnsmit) [inline scripts/aura.py:32]: specs as given by @Tritium-VLK clearly ask for all numbers adding up to 1., so i would drop the `* 100` here.

**Review by jorijnsmit — COMMENTED** [review]
- im not too familiar with the rewards' internal methods such as `get_bvecvx_lp_ppfs` or `fuse_snapshot_of_token`, but taking those at face value. 
- ~~@Tritium can you confirm aura snapshot is at block 14829454?~~ confirmed: https://discord.com/channels/785315893960900629/868110311390064681/979819438364848138
- list of 15 beneficiaries for ftoken checks out against amount of ftoken holders

File: scripts/aura.py (hunk 5 above)
Comment (SHAKOTN) [reply scripts/aura.py]: Done

**Review by jorijnsmit — APPROVED** [review]
having permission issues in regards to accessing dynamodb, so cant run locally.

but changes lgtm, although threshold might still be too low. difficult to say without more input from aura.

**Pull request merged**